src/BufferedVectors.jl

```diff
@@ -29,6 +29,7 @@ Base.IteratorSize(::BufferedVector) = Base.HasLength()
 Base.IteratorEltype(::BufferedVector) = Base.HasEltype()
 Base.collect(x::BufferedVector{T}) where {T} = length(x) > 0 ? @inbounds(x.elements[1:length(x)]) : T[]
 Base.eltype(::BufferedVector{T}) where T = T
+@inline Base.push!(buffer::T, x::S) where {T,S} = push!(buffer, convert(T, x))
```

Comment:
out of interest, why not just
```suggestion
@inline Base.push!(buffer::T, x) where {T} = push!(buffer, convert(T, x))
```

Reply:
I just like the explicitness of the two method signatures:
```julia
Base.push!(buffer::T, x::S) where {T,S}
Base.push!(buffer::T, x::T) where {T}
```
I'd like to keep it if that's ok